### PR TITLE
rkt: Pass through podIP

### PR DESCRIPTION
**Pull request by euank**
This is needed for the /etc/hosts mount and the downward API to work.
Furthermore, this is required for the reported `PodStatus` to be
correct.

The `Status` bit mostly worked prior to #25062, and this restores that
functionality in addition to the new functionality.

In retrospect, the regression in status is large enough the prior PR should have included at least some of this; my bad for not realizing the full implications there.
#25902 is needed for downwards api stuff, but either merge order is fine as neither will break badly by itself.

cc @yifan-gu @dcbw

### CODE REVIEW

File: pkg/kubelet/rkt/rkt.go [hunk 1]
```diff
@@ -2054,13 +2060,10 @@ func (r *Runtime) GetPodStatus(uid types.UID, name, namespace string) (*kubecont
...
 
-	if latestPod != nil {
-		// Try to fill the IP info.
-		for _, n := range latestPod.Networks {
-			if n.Name == defaultNetworkName {
-				podStatus.IP = n.Ipv4
-			}
-		}
+	// TODO(euank): this will not work in host networking mode
+	containerID := kubecontainer.ContainerID{ID: string(uid)}
```
Comment (yifan-gu) [inline pkg/kubelet/rkt/rkt.go]: Just to be clean, so in the host network, the pod ip is not set here, but will be set in https://github.com/kubernetes/kubernetes/pull/24633/files , right?

**Comment by yifan-gu** [comment]
LGTM

File: pkg/kubelet/rkt/rkt.go (hunk 1 above)
Comment (euank) [inline pkg/kubelet/rkt/rkt.go]: It looks like it actually does get set there, yeah. Maybe this should be a `note`, not a `TODO`, I'll do some quick experimenting.

**Comment by euank** [comment]
After looking a bit further (thanks @yifan-gu), #25902 is needed, alongside this, before things work correctly.

**Comment by dcbw** [comment]
lgtm

**Comment by euank** [comment]
@k8s-bot test this issue: #26302

**Pull request merged**